sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_providers/__init__.py

```diff
@@ -8,22 +8,28 @@
 from .local_provider import LocalCryptographyProvider
 from .rsa import RsaCryptographyProvider
 from .symmetric import SymmetricCryptographyProvider
-from ... import KeyType
+from ... import JsonWebKey, KeyType
 
 if TYPE_CHECKING:
+    from typing import Union
     from ... import KeyVaultKey
 
 
 def get_local_cryptography_provider(key):
-    # type: (KeyVaultKey) -> LocalCryptographyProvider
-    if key.key_type in (KeyType.ec, KeyType.ec_hsm):
+    # type: (Union[JsonWebKey, KeyVaultKey]) -> LocalCryptographyProvider
```

Comment:
I also thought about whether we should just use the JsonWebKey for local crypto, and the only snag I can think of is that a JsonWebKey's `kid` could be different from a KeyVaultKey's `id`. It seems unlikely that someone would intentionally assign those properties different values to some end, but it's not impossible

Reply:
This could probably be resolved by passing a keyword argument corresponding to KeyVaultKey.id like `_key_id` to get_local_cryptography_provider, and preferring that value over the JsonWebKey's `kid` in LocalCryptographyProvider.__init__